internal/pkg/store/ssm/ssm.go

```diff
@@ -231,10 +215,109 @@ func (s *SSM) ListEnvironments(projectName string) ([]*archer.Environment, error
+	serializedEnvs, err := s.listParams(environmentsPath)
+	if err != nil {
+		return nil, err
+	}
+	for _, serializedEnv := range serializedEnvs {
+		var env archer.Environment
+		if err := json.Unmarshal([]byte(*serializedEnv), &env); err != nil {
+			return nil, err
+		}
+
+		environments = append(environments, &env)
+	}
+	return environments, nil
+}
+
+// CreateApplication instanciates a new application within an existing project. Returns ErrApplicationAlreadyExists
+// if the application already exists in the project.
+func (s *SSM) CreateApplication(app *archer.Application) error {
+	if _, err := s.GetProject(app.Project); err != nil {
+		return err
+	}
+
+	applicationPath := fmt.Sprintf(fmtAppParamPath, app.Project, app.Name)
+	data, err := marshal(app)
+	if err != nil {
+		return err
+	}
+
+	paramOutput, err := s.systemManager.PutParameter(&ssm.PutParameterInput{
+		Name:        aws.String(applicationPath),
+		Description: aws.String(fmt.Sprintf("ECS-CLI Application %s", app.Name)),
+		Type:        aws.String(ssm.ParameterTypeString),
+		Value:       aws.String(data),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ssm.ErrCodeParameterAlreadyExists:
+				return &store.ErrApplicationAlreadyExists{
+					ApplicationName: app.Name,
+					ProjectName:     app.Project}
+			}
+		}
+		return err
+	}
+
+	log.Printf("Created Application with version %v", *paramOutput.Version)
+	return nil
+
+}
+
+// GetApplication gets an application belonging to a particular project by name. If no app is found
+// it returns ErrNoSuchApplication.
+func (s *SSM) GetApplication(projectName, appName string) (*archer.Application, error) {
+	applicationPath := fmt.Sprintf(fmtAppParamPath, projectName, appName)
+	applicationParam, err := s.systemManager.GetParameter(&ssm.GetParameterInput{
+		Name: aws.String(applicationPath),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if applicationParam.Parameter.Value == nil {
+		return nil, &store.ErrNoSuchApplication{
+			ProjectName:     projectName,
+			ApplicationName: appName,
+		}
+	}
+
+	var app archer.Application
+	err = json.Unmarshal([]byte(*applicationParam.Parameter.Value), &app)
+	return &app, err
+}
+
+// ListApplications returns all applications belonging to a particular project.
+func (s *SSM) ListApplications(projectName string) ([]*archer.Application, error) {
+	var applicaitons []*archer.Application
+
+	applicaitonsPath := fmt.Sprintf(rootAppParamPath, projectName)
+	serializedApps, err := s.listParams(applicaitonsPath)
+	if err != nil {
+		return nil, err
+	}
+	for _, serializedApp := range serializedApps {
+		var app archer.Application
+		if err := json.Unmarshal([]byte(*serializedApp), &app); err != nil {
+			return nil, err
+		}
+
+		applicaitons = append(applicaitons, &app)
+	}
+	return applicaitons, nil
+}
+
+func (s *SSM) listParams(path string) ([]*string, error) {
+	var stringParams []*string
```

Comment:
nit: Can we avoid repeating the type in the variable's name? maybe `rawParams` or `unserializedParams`

Reply:
👍